### ISSUE TITLE
update for 2024 protocol

### ISSUE THREAD

**Issue by Controls-Cat5e** [Boomaa23/open-ds#23]

**Comment by Boomaa23** [comment]
Hi there! I don't have access to a roborio or working robot regularly and as such may not be able to update the software. I am looking into purchasing a rio but that is a significant expense for me as an individual.

**Comment by ZenoCoding** [comment]
This is a great project, but the lack of support for newer versions makes it impractical. I have access to multiple roboRIOs and working robots. Do you think I could help? I'm still a student, and considering that it is competition season, I don't have a lot of free time, but I'd be happy to do some testing.

**Comment by Boomaa23** [comment]
@ZenoCoding absolutely! I'm pretty busy myself, but during the summer I should be able to put more time into development. If you'd like to test things now or then please go ahead; everything should be linked on the README or in the source code somewhere. You can reach me at my email linked on my Github profile if you want some more direct correspondence.

**Comment by ethanc8** [comment]
Hi! In my testing version 0.2.4 worked on the 2024 protocol. I don't know if anything will change for the 2025 season, though. The radio has significantly changed ([we are now using VH-109](https://frc-radio.vivid-hosting.net/)), but I don't think that should affect the protocol.

**Comment by ZenoCoding** [comment]
Sorry for my late response! It's been a hell of a competition season, and back then I was just building the kitbot!

In my testing, everything worked great in terms of robot connection. I think in my original tests I had forgotten to add my team number 😅 

Still, I am unable to connect a controller or detect it when connecting the FRC kit of parts controllers via USB-C (with an adapter). Is this a mac issue? I'm using an M2 mac.

**Comment by Boomaa23** [comment]
Yeah, that's a mac issue that I've patched in a commit but never got around to making a release for. Try using this jar (it's just a build of the latest commit) https://opends.aptapus.net/builds/a796a01.jar

**Comment by ZenoCoding** [comment]
Interestingly, I can now connect the controller but can't seem to connect to the robot. Any ideas as to what could be causing this? There are no errors in the console...
<img width="727" alt="Screenshot 2024-10-19 at 12 18 39 AM" src="https://github.com/user-attachments/assets/8f4f0e0f-cc6f-477d-8db7-48ae5d56d693">

**Comment by ZenoCoding** [comment]
I just remembered that I was connected via Ethernet last time. I will test that setup tomorrow when I get my hands on the robot again. My guess is that wifi connection didn't work then, either.
